### DISCUSSION TITLE
Hue Wall Switch Dual Rocker Multiple Events

### DISCUSSION THREAD

**Discussion by jamesonuk** [Koenkk/zigbee2mqtt#21640] [Devices]
Is anyone using the Hue Wall Switch module (https://www.zigbee2mqtt.io/devices/929003017102.html#philips-929003017102) in dual rocker mode?  @Koenkk you are using one of these right?

I use several of these devices bound to groups and they work fine but I have one switch downstairs which has two switches so I am using it in normal mode and handling the left / right press events in HA.

The issue is pressing the switch is spamming events 
```
	Line 11: Info 2024-03-01 19:33:02MQTT publish: topic 'zigbee2mqtt/Hall Hue Wall Switch 2/action', payload 'left_press'
	Line 22: Info 2024-03-01 19:33:03MQTT publish: topic 'zigbee2mqtt/Hall Hue Wall Switch 2/action', payload 'left_press'
	Line 33: Info 2024-03-01 19:33:03MQTT publish: topic 'zigbee2mqtt/Hall Hue Wall Switch 2/action', payload 'left_press'
	Line 44: Info 2024-03-01 19:33:03MQTT publish: topic 'zigbee2mqtt/Hall Hue Wall Switch 2/action', payload 'left_press'
	Line 55: Info 2024-03-01 19:33:04MQTT publish: topic 'zigbee2mqtt/Hall Hue Wall Switch 2/action', payload 'left_press'
	Line 66: Info 2024-03-01 19:33:04MQTT publish: topic 'zigbee2mqtt/Hall Hue Wall Switch 2/action', payload 'left_press'
	Line 77: Info 2024-03-01 19:33:07MQTT publish: topic 'zigbee2mqtt/Hall Hue Wall Switch 2/action', payload 'left_press'
	Line 88: Info 2024-03-01 19:33:07MQTT publish: topic 'zigbee2mqtt/Hall Hue Wall Switch 2/action', payload 'left_press'
```
and that is just the events I grabbed from the log and I think there were a few more.

This is making the switch unusable as the lights are flickering on and off several times and sometimes end up on and sometimes off.

I am also seeing random events turn up hours later (this is a much cut down version of what happened last night) but I turned lights off around 9pm but they then flicked on and off at 00:51 (just once) an then at 01:11 I got the on/off actions four times followed by an on action.   I woke up and turned the light group over via HA at 01:49 then at 02:36 I got something like 28 on/off actions, at 02:56 I got two on/off actions and then an on leaving them on.   I turned them off via HA and then no more events until I turned the lights on in the morning.

![image](https://github.com/Koenkk/zigbee2mqtt/assets/1040621/23ad5c72-e619-4e2c-84ab-45db80e07ca1)

I have checked all the wiring and even replaced the Hue Switch module in case it was faulty but still get the same.

Anyone also using them as dual rocker (or could at least try) ?

**Comment by jamesonuk** [comment]
Just to add this is left_press but right_press works fine (I turned switch on and then off and only get two events fired)
![image](https://github.com/Koenkk/zigbee2mqtt/assets/1040621/8334ca0e-7ea0-4ad6-9e6d-e99d4ea7157a)

**Comment by Koenkk** [comment]
Not using them as dual (single with binding to a group)

**Comment by imchasingshadows** [comment]
I'm also using these modules and I'm looking for a solution to use double rockers when the coordinator is offline. @jamesonuk did you ever find a solution for this?

**Reply by jamesonuk** [reply]
Binding can only be done with one channel (there is only one on/off cluster) so you can't use them with two switches in that way (only through automations)

**Reply by imchasingshadows** [reply]
Alright thanks, too bad... 

It's kinda stupid because these modules aren't cheap and almost everywhere i use them i'd have to get a second one to support the second switch :\

**Reply by jamesonuk** [reply]
You can still use them in automations (which is what the Hue hub must do).   I do that in a couple of places but being able to bind them is obviously a massive plus.

One thing to note is that Phillips have broken the binding in recent versions anyway.   They now send alternate on/off commands rather than toggle.   This means they don't work properly via binding anyway

**Reply by imchasingshadows** [reply]
Automations is how they currently work for me. I use node-red to interpret the messages sent, and then use those messages to toggle the lights on/off. This works pretty well. Though problems arise when homeassitant/z2m/nodered goes offline. 

In my opinion its not really a solution to "just wait for the stack to come online" -- I want my lights to turn on right now and not in 1 min.

So that's why i'm looking into a solution that allows me to bind the switch directly to a light(group).

Regarding the broken on/off commands i think im experiencing this with the [remotes](https://www.zigbee2mqtt.io/devices/929002398602.html), not with the wall switches (?)

**Reply by jamesonuk** [reply]
Yes I believe they have made the newer models work like the remotes https://github.com/Koenkk/zigbee2mqtt/issues/26775
It isn't clear whether it is the RDM001 vs RDM004 or firmware v1 vs v2 that makes this change but certainly I bought another one late last year and it now send on/off like the remote rather than a toggle like my older (RDM001 / v1 firmware) ones do which makes them useless for binding
